sdk/nodejs/query/operators.ts

```diff
@@ -0,0 +1,746 @@
+                yield t;
+            } else if (predicate(t, i) === false) {
+                stopSkipping = true;
+                yield t;
+            }
+        }
+    };
+}
+
+export function take<TSource>(n: number): Operator<TSource, TSource> {
+    if (n < 0) {
+        throw Error("take was provided a negative number of elements to take");
+    }
+
+    return async function*(source: AsyncIterableIterator<TSource>) {
+        for await (const [t, i] of zip(source, range(0))) {
+            if (i >= n) {
+                return;
+            }
+            yield t;
+        }
+    };
+}
+
+export function takeWhile<TSource>(
+    predicate: (t: TSource, i: number) => boolean,
+): Operator<TSource, TSource> {
+    return async function*(source: AsyncIterableIterator<TSource>) {
+        for await (const [t, i] of zip(source, range(0))) {
+            if (predicate(t, i) === false) {
+                return;
+            }
+            yield t;
+        }
+    };
+}
+
+//
+// Join operators.
+//
+
+export function join<TOuter, TInner, TKey, TResult>(
+    inner: AsyncIterableIterator<TInner>,
+    outerKeySelector: (to: TOuter) => TKey,
+    innerKeySelector: (ti: TInner) => TKey,
+    resultSelector: (to: TOuter, ti: TInner) => TResult,
+): Operator<TOuter, TResult> {
+    return async function*(outer: AsyncIterableIterator<TOuter>) {
+        const inners = new Map<TKey, TInner>();
+
+        for await (const t of inner) {
+            inners.set(innerKeySelector(t), t);
+        }
+
+        for await (const t of outer) {
+            const key = outerKeySelector(t);
+            if (key === undefined) {
+                continue;
+            } else if (inners.has(key)) {
+                yield resultSelector(t, <TInner>inners.get(key));
+            }
+        }
+    };
+}
+
+// public static IEnumerable<TResult> GroupJoin<TOuter, TInner, TKey,
+// TResult>(
+//     this IEnumerable<TOuter> outer,
+//     IEnumerable<TInner> inner,
+//     Func<TOuter, TKey> outerKeySelector,
+//     Func<TInner, TKey> innerKeySelector,
+//     Func<TOuter, IEnumerable<TInner>, TResult> resultSelector);
+// public static IEnumerable<TResult> GroupJoin<TOuter, TInner, TKey,
+// TResult>(
+//     this IEnumerable<TOuter> outer,
+//     IEnumerable<TInner> inner,
+//     Func<TOuter, TKey> outerKeySelector,
+//     Func<TInner, TKey> innerKeySelector,
+//     Func<TOuter, IEnumerable<TInner>, TResult> resultSelector,
+//     IEqualityComparer<TKey> comparer);
+
+//
+// Concatenation operators.
+//
+
+export function concat<TSource>(iter: AsyncIterable<TSource>): Operator<TSource, TSource> {
+    return async function*(source: AsyncIterableIterator<TSource>) {
+        for await (const t of source) {
+            yield t;
+        }
+
+        for await (const t of iter) {
+            yield t;
+        }
+    };
+}
+
+//
+// Ordering operators.
+//
+
+// public static OrderedSequence<TSource> OrderBy<TSource, TKey>(
```

Comment:
Can we remove all of these comments?  I understand that they are here because we are mimicking the LINQ surface area, but I find them very confusing in the context of typescript files.